### PR TITLE
Auto qlog

### PR DESCRIPTION
**Pull request by huitema**

### CODE REVIEW

File: picoquic/logwriter.c [hunk 1]
```diff
@@ -746,7 +748,8 @@ FILE* create_binlog(char const* binlog_file, uint64_t creation_time);
 
 void binlog_new_connection(picoquic_cnx_t * cnx)
 {
-    if (cnx == NULL || cnx->quic->binlog_dir == NULL) {
+    char const* bin_dir = (cnx->quic->binlog_dir == NULL) ? cnx->quic->qlog_dir : cnx->quic->binlog_dir;
+    if (cnx == NULL || bin_dir == NULL) {
```
Comment (steschu77) [inline picoquic/logwriter.c:752]: If cnx was NULL then you'd have crashed the line before already :)

File: picoquic/logwriter.c [hunk 2]
```diff
@@ -822,6 +829,42 @@ void binlog_close_connection(picoquic_cnx_t * cnx)
     fflush(f);
 
     cnx->f_binlog = picoquic_file_close(cnx->f_binlog);
+
```
Comment (steschu77) [inline picoquic/logwriter.c:832]: Would it be possible to have a #define here such that you can exclude the QLOG stuff from picoquic for people who like the "pico" in picoquic? :) This also includes all the additional files you added to the picoquic core library for the QLOG conversion.

File: picoquic/logwriter.c (hunk 1 above)
Comment (huitema) [reply picoquic/logwriter.c]: Removed the extra test. It would mess up static analysis.

File: picoquic/logwriter.c (hunk 2 above)
Comment (huitema) [reply picoquic/logwriter.c:830]: I did not use a #define. Instead, I used a function pointer. I moved the files back to the log library, and added there an autoqlog file that defines `picoquic_set_qlog` and documents the function pointer. Applications that want to use that API will have to link the log library, and those that do without get a smaller code size.

Comment (huitema) [reply picoquic/logwriter.c:830]: @steschu77 I think that fixes the code size issue that you mentioned. Will further reduce it once we get rid of the text log.

**Comment by huitema** [comment]
Merging so I can prepare the setupfor the interop on 7/14.

File: picoquic/logwriter.c (hunk 2 above)
Comment (steschu77) [reply picoquic/logwriter.c:830]: Thanks, @huitema ! That is exactly what I was looking for!

**Pull request merged**